File: test/integration_test.go
package main

import (
	"os"
	"os/exec"
	"testing"
)

func TestGoAndPythonIntegration(t *testing.T) {
	// Step 1: Generate sample data using the Python script
	cmd := exec.Command("python", "scripts/create_sample_data.py")
	err := cmd.Run()
	if err != nil {
		t.Fatalf("Failed to generate sample data: %v", err)
	}

	// Step 2: Process the data using the Python script
	cmd = exec.Command("python", "scripts/process_data.py")
	err = cmd.Run()
	if err != nil {
		t.Fatalf("Failed to process data: %v", err)
	}

	// Step 3: Run the Go application to categorize the descriptions
	cmd = exec.Command("go", "run", "cmd/main.go")
	err = cmd.Run()
	if err != nil {
		t.Fatalf("Failed to run Go application: %v", err)
	}

	// Step 4: Verify that the output file is generated
	if _, err := os.Stat("data/processed_descriptions.xlsx"); os.IsNotExist(err) {
		t.Fatalf("Output file processed_descriptions.xlsx does not exist")
	}
}

File: test/integration_test.py
import unittest
import os
import subprocess

class TestIntegration(unittest.TestCase):

    def test_integration(self):
        # Step 1: Generate sample data using the Python script
        result = subprocess.run(['python', 'scripts/create_sample_data.py'], capture_output=True, text=True)
        self.assertEqual(result.returncode, 0, f"Error: {result.stderr}")

        # Step 2: Process the data using the Python script
        result = subprocess.run(['python', 'scripts/process_data.py'], capture_output=True, text=True)
        self.assertEqual(result.returncode, 0, f"Error: {result.stderr}")

        # Step 3: Verify that the output file is created
        self.assertTrue(os.path.exists('data/processed_descriptions.xlsx'))

if __name__ == '__main__':
    unittest.main()
